modules/affile/wildcard-source.c

```diff
@@ -77,11 +77,24 @@ _remove_file_reader(FileReader *reader, gpointer user_data)
     }
   log_pipe_unref(&reader->super);
 
-  gchar *full_path = pending_file_list_pop(self->waiting_list);
-  if (full_path)
+  gchar *full_path = NULL;
+  GQueue *q = g_queue_new();
```

Comment:
This has to be freed with `g_queue_free`.

Reply:
Just a thought (completely optional):
I was thinking of a cleaner implementation, based on `g_queue_peek_nth()` or `g_queue_peek_head_link()`, or with a simple GList reimplementation, but `pending_file_list` would require an iterator interface with cancellation.

For example:
```
full_path = pending_file_find_and_steal(self->waiting_list, FindFunc);
```
or
```
while ((full_path_glist = pending_file_list_next(self->waiting_list)))
{
  if (...)
    pending_file_list_unlink(full_path_glist)
}
```